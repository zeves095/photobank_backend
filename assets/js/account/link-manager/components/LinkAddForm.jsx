import React, {Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {submitLink, updateLink} from '../actionCreator';
import FormWrapper from '../../../forms/FormWrapper';
import {getLinkTargets, getChosenResource} from '../selectors';
import {Confirmator} from '../../../common/Confirmator';
import {validateLinkAddForm} from '../../../common/utils/validation';

/**
 * Форма для добавления ссылки
 */
export class LinkAddForm extends React.Component {
  /**
   * Конструктор компонента
   * form_error - Текст ошибки формы
   * defaults - Данные формы по умолчанию
   * confirmatorQuestions - вопросы для компонента Confirmator
   *
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props) {
    super(props);
    this.state={
      form_error: false,
      defaults:{},
      confirmatorQuestions:[
        "Вы уверены?"
      ]
    };
  }

  /**
   * Делает проверку на факт существования выбранного ресурса на момент маунта выбранного ресурса
   */
  componentDidMount(){
    if(this.props.resource_chosen !== null){
      this.handleSetResource();
    }
  }

  /**
   * Делает проверку на факт изменения выбранного ресурса
   */
  componentDidUpdate(prevProps){
    if(prevProps.resource_chosen !== this.props.resource_chosen){
      this.handleSetResource();
    }
  }

  /**
   * Проставляет поле resource для формы из данных redux
   */
  handleSetResource = ()=>{
    let defaults = this.state.defaults;
    let resources = this.props.resource_chosen.map((resource)=>{return resource.id}).join(',');
    defaults.resource = resources;
    this.setState({
      defaults
    });
  }

  /**
   * Обработчик изменений формы. Обновляет состояние формы после изменения поля
   * @param  {Object} data Данные формы
   */
  handleInputChange = (data)=>{
    this.setState({
      defaults:data
    });
  }

  /**
   * Обработчик отправки формы. Отправляет action с данными формы для создания ссылки
   * @param  {Object} data Данные формы
   */
  handleFormSubmit = (data) =>{
    this.props.submitLink(data);
  }

  /**
   * Обработчик ошибок формы.
   * @param  {string[]} errors [description]
   */
  handleFormError = (errors) =>{
    this.setState({
      form_error: errors.length>0
    });
  }

  /**
   * Обработчик потери фокуса формы. Сейчас не испоьзуется, но пусть будет пока.
   * @param  {Object} data Данные формы
   */
  handleFormBlur = (data)=>{
    //this.props.validateLinkAddForm(data);
  }

  /**
   * Создает список вопросов для подтвердждения в компоненте Confirmator. Проверяет, пытается ли юзер создать ссылки для ресурсов, у которых уже есть ссылки, либо добавить ссылки в уже существующую группу
   */
  getConfirmatorQuestions= ()=>{
    let defaults = this.state.defaults;
    let confirmatorQuestions = this.state.confirmatorQuestions;
    if(typeof defaults.target !== "undefined"){
      if(this.props.targets.includes(defaults.target)){
        let existing = this.props.resource_chosen.filter((res)=>{return typeof res.link_exists !== 'undefined' && res.link_exists && res.link_targets.includes(defaults.target)});
        if(existing.length>0){
          confirmatorQuestions[2] = existing.length+" ресурсов уже имеют ссылки. Все равно продолжить?";
        }else{

          confirmatorQuestions.splice(2, 1);
        }
        confirmatorQuestions[1] = "Группа с именем "+defaults.target+" уже существует. Все равно добавить?";
      }else{
        confirmatorQuestions.splice(1, 1);
        confirmatorQuestions.splice(2, 1);
      }
    }
    this.setState({
      confirmatorQuestions
    });
  }

  render() {
    let confirmator = (<Confirmator active={this.props.resource_chosen.length>0} questions={this.state.confirmatorQuestions} prehook={this.getConfirmatorQuestions} onConfirm={()=>{}} inline={false} disabled={this.state.form_error} buttonTitle={"Добавить"} />);
    return (
      <div className={"link-add-form "+(this.props.resource_chosen.length>0?"open":"")}>
        {this.props.form_error !== null?<div className="form-error">{this.props.form_error}</div>:null}
        <FormWrapper form="link-add" onChange={this.handleInputChange} validate={validateLinkAddForm} onBlur={this.handleFormBlur} onSubmit={this.handleFormSubmit} onError={this.handleFormError} defaults={this.state.defaults} submit={confirmator} />
      </div>);
  }

}

const mapStateToProps = (state) =>{
  return {
    link_editing: state.link.link_editing,
    link: state.link.link_editing_id,
    resource_chosen: getChosenResource(state),
    targets: getLinkTargets(state),
    form_error: state.ui.form.link_add.error
  }
}

const mapDispatchToProps = {
  submitLink,
  updateLink,
  //validateLinkAddForm
}

export default connect(mapStateToProps, mapDispatchToProps)(LinkAddForm);
