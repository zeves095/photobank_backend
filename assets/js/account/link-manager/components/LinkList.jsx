import React, {Component} from 'react';
import {connect} from 'react-redux';
import { getLinkTargets } from '../selectors';
import { chooseLink, addLink, fetchLinks, deleteLink, stopEditing } from '../actionCreator'
import {NotificationService} from '../../../services/NotificationService';
import {ModalImage} from '../../../common/ModalImage';
import {Confirmator} from '../../../common/Confirmator';
/**
 * Список существующих ссылок для текущего пользователя
 */
export class LinkList extends React.Component{
  /**
   * Конструктор компонента
   * links - Существующие ссылки для отображения
   * target - Целевая группа ссылок
   * modal_image_url - Ссылка на изображение для отображения в модальном окне предпросмотра
   * confirmatorQuestions - вопросы для компонента Confirmator
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props){
    super(props);
    this.state={
      links:this.props.links,
      target:"Все",
      modal_image_url:"",
      confirmatorQuestions:[
        "Вы уверены?"
      ],
    };
    this.eventHandler = {};
  }

  /**
   * Фильтрует ссылки
   */
  componentDidUpdate(prevProps, prevState){
    if(prevProps.links !== this.props.links || prevState.f !== this.state.f ){
      this.setState({
        links:this.props.links.filter((item)=>{if(!this.state.f) return true; return JSON.stringify(item).includes(this.state.f);})
      });
    }
  }

  /**
   * Обработчик клика на ссылку. Пока не используется, в будущем планируется редактировать выбранную ссылку по клику
   * @param  {Number} id Идентификатор ссылки
   */
  handleLinkClick = (id)=>{
    //this.props.chooseLink(id);
  }

  /**
   * Обработчик добавления новой ссылки
   * @param  {Event} e Событие клика
   */
  handleLinkAdd = (e)=>{
    this.props.editing||this.props.adding?this.props.stopEditing():this.props.addLink();
  }

  /**
   * Обработчик клика но группу ссылок
   * @param  {Event} e Событие клика
   */
  handleTargetChoice = (e)=>{
    e.preventDefault();
    this.setState({
      target:e.target.dataset['target']
    });
  }

  /**
   * Обработчик удаления ссылки
   * @param  {Number} id Идентификатор ссылки для удаления
   */
  handleLinkDelete = (id)=>{
    this.props.deleteLink(id);
  }

  /**
   * Обработчик копирования всех отображаемых ссылок в буфер обмена
   */
  handleCopyAllToClipboard = ()=>{
    let links = [];
    this.state.links.forEach((link)=>{
      if(link.target !== this.state.target && this.state.target !== "Все"){return false;}
      links.push('https://photobank.domfarfora.ru'+link.external_url);
    });
    if(typeof navigator.clipboard !== "undefined"){navigator.clipboard.writeText(links.join(",\n"))}else{NotificationService.throw('clipboard-error')};
    NotificationService.toast("link-copied");
  }

  /**
   * Обработчик скачивания txt-файла с текстом всех отображаемых ссылок
   */
  handleGetTxt =()=>{
    let links = this.state.links
    .filter((link)=>{
        return link.target == this.state.target || this.state.target == "Все";
    })
    .map((link)=>{
        return(link.link_id);
      }
    );
    if(links.length === 0){NotificationService.toast("custom", "Нет ссылок"); return;}
    let linkTxtForm = document.createElement("form");
    linkTxtForm.target = "_blank";
    linkTxtForm.method = "POST";
    linkTxtForm.action = "/api/links/txt/";
    let linkTxtInput = document.createElement("input");
    linkTxtInput.type = "text";
    linkTxtInput.name = "links";
    linkTxtInput.value = links;
    linkTxtForm.appendChild(linkTxtInput);
    document.body.appendChild(linkTxtForm);
    linkTxtForm.submit();
    linkTxtForm.remove();
  }

  /**
   * Обработчик открытия модального окна с превью изображения ресурса
   * @param  {Number} link Идентификатор ссылки
   */
  handleModalImage = (link)=>{
    this.setState({
      modal_image_url: link
    });
  }

  /**
   * Обработчик закрытия модального окна
   */
  handleModalClose = ()=>{
    this.setState({
      modal_image_url: ""
    });
  }

  /**
   * Подписчик на события на странице для отлавливания комбинаций клавиш
   * @param  {Event} e Событие
   */
  listener = (e)=>{
    e = e || window.event;
    var key = e.which || e.keyCode; // keyCode detection
    var ctrl = e.ctrlKey ? e.ctrlKey : ((key === 17) ? true : false); // ctrl detection

    if ( key == 67 && ctrl ) {
        this.handleCopyAllToClipboard();
      }else if ( key == 83 && ctrl ) {
        e.preventDefault();
        this.handleGetTxt();
        return false;
      }
  }

  /**
   * Обработчик появления курсора мыши на области компонента, для ограничения событий нажатия клавиатурных сокращений
   */
  handleFocus = ()=>{
    document.body.addEventListener("keydown",this.listener);
  }

  /**
   * Обработчик исчезновения курсора мыши с области компонента, для ограничения событий нажатия клавиатурных сокращений
   */
  handleBlur = ()=>{
    document.body.removeEventListener('keydown', this.listener);
  }

  render(){
    let filter = <input type="search" className="link-filter" onChange={(e)=>{this.setState({f : e.target.value})}}  placeholder="Фильтр"/>
    let links = this.state.links.map(
      (link)=>{
        if(link.target !== this.state.target && this.state.target !== "Все"){return false;}
        let delete_btn = <i className="fas fa-trash-alt"></i>;
        let thumb = this.props.thumbs.find((thumb)=>thumb.id === link.resource_id);
        return(
          <div data-linkid={link.link_id} key={"link"+link.link_id} className="link " onClick={()=>{this.handleLinkClick(link.link_id)}}>
            <Confirmator questions={this.state.confirmatorQuestions} prehook={()=>{}} active={true} onConfirm={()=>{this.handleLinkDelete(link.link_id)}} inline={true} customClass={"delete-link"} disabled={false} buttonTitle={delete_btn} />
            <div className="link-info">
            <div><b>Ссылка:</b><a href={'https://photobank.domfarfora.ru'+link.external_url} target="_blank" >{'https://photobank.domfarfora.ru'+link.external_url}</a></div>
          <div><b>Товар: </b>{link.item_name}({link.item_id})</div>
          </div>
        <span className={"resource-preview"+(typeof thumb === 'undefined'?" resource-preview--loading":"")} style={{backgroundImage:typeof thumb === 'undefined'?"none":"url(/catalogue/node/item/resource/"+thumb.thumb_id+".jpg)"}} onClick={(e)=>{e.stopPropagation();this.handleModalImage("/catalogue/node/item/resource/"+thumb.id+".jpg")}}></span>
          </div>
        )
      }
    );
    let tabs = ["Все"].concat(this.props.targets).map((target)=>{
        return(
          <button key={"btn-"+target} className={(target===this.state.target?" active":"")} data-target={target} onClick={this.handleTargetChoice}>{target}</button>
        )
    })
    return(
      <div className={"link-list "+(this.props.adding||this.props.editing?"shrunk":"")} onMouseEnter={this.handleFocus} onMouseLeave={this.handleBlur}>
        {this.state.modal_image_url !== ""?<ModalImage image_url={this.state.modal_image_url} closeModalHandler={this.handleModalClose} />:null}
        <div className="component-header">
          <h2 className="component-title">
            Ссылки
          </h2>
          {filter}
        </div>
        <div className={"component-body" + (this.props.loading?" loading":"")}>
          <div className="component-body__top-section">
            <div className="button-block">
              <button key="btn-add" onClick={this.handleLinkAdd} style={{float:"none"}} className=" waves-effect hoverable waves-light btn add-button" type="button">{this.props.adding||this.props.editing?(<span><i className="fas fa-ban"></i>Отмена</span>):(<span><i className="fas fa-plus-circle"></i>Добавить</span>)}</button>
            <button key="btn-copy" onClick={this.handleCopyAllToClipboard} style={{float:"none"}} className=" waves-effect hoverable waves-light btn add-button" type="button"><i className="fas fa-copy"></i>Скопировать в буфер {!this.props.adding&&!this.props.editing&&(<span className="help-text">&lt;Ctrl&gt;+&lt;C&gt;</span>)}</button>
          <button key="btn-dl" onClick={this.handleGetTxt} style={{float:"none"}} className=" waves-effect hoverable waves-light btn add-button" type="button"><i className="fas fa-align-justify"></i>Скачать ссылки {!this.props.adding&&!this.props.editing&&(<span className="help-text">&lt;Ctrl&gt;+&lt;S&gt;</span>)}</button>
            </div>
            <div className="link-list__tabs button-block">
              {tabs}
            </div>
            {links.length==0?(<div className="resource plaque warning"><i className="fas fa-times-circle"></i>Нет ссылок</div>):links}
          </div>
        </div>
      </div>
    );
  }

}

const mapStateToProps = (state) =>{
  return {
    links: state.link.links_done,
    editing: state.link.link_editing,
    adding: state.link.link_adding,
    targets: getLinkTargets(state),
    thumbs: state.resource.resources_thumbnails,
    loading: state.ui.loading.link_list
  }
}

const mapDispatchToProps = {
    chooseLink,
    addLink,
    fetchLinks,
    deleteLink,
    stopEditing
}

export default connect(mapStateToProps, mapDispatchToProps)(LinkList);
