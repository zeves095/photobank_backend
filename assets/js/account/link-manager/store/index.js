import React from 'react';
import {createStore, combineReducers, applyMiddleware, compose} from 'redux';
import thunk from 'redux-thunk';
import reducer from './reducers';
import api from './middleware/api';

const composeEnhancers =
  typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose

let enhancer = composeEnhancers(
  applyMiddleware(thunk)
);

export const store = createStore(
  reducer,enhancer
);
