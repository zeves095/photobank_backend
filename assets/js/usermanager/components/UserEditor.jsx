import React from 'react';
import {connect} from 'react-redux';
import {submitUser} from '../actionCreator';
import selectors from '../selectors';
/**
 * Компонент для отображения интерфейса редактирования или создания пользователя
 */
export class UserEditor extends React.Component {

  /**
   * Конструктор компонента
   * user - Объект текущего пользователя
   * sent - Была ли отправлена форма с этими данными
   * hide_password - Отображать ли пароль как plaintext или скрывать
   */
  constructor(props) {
    super(props);
    this.state ={
      "sent": false,
      "hide_password": true
    }
  }

  /**
   * Обработчик отправки формы с данными пользователя
   * @param  {Event} e Событие клика
   */
  handleSubmit=(e)=>{
    e.preventDefault();
    let newUser={
      "id":this.refs["id-input"].value,
      "password":this.refs["password-input"].value,
      "email":this.refs["email-input"].value,
      "active":this.refs["active-input"].checked,
      "name":this.refs["name-input"].value,
      "role":this.refs["role-input"].value,
    }
    this.props.submitUser(newUser);
    this.setState({
      "sent":true
    });
  }

  /**
   * Скрывает/открывает поле пароль
   * @param  {Event} e Событие клика
   */
  hidePassword=(e)=>{
    e.preventDefault();
    this.setState({
      "hide_password": !this.state.hide_password
    });
  }

  componentDidUpdate(prevProps, prevState){
    if(prevState.sent == true && this.props.user.id != prevProps.user.id){
      this.setState({
        "sent":false
      });
    }
  }

  render() {
    if(this.props.user == null){return (
      <div className="user-editor col s8" key={"user"+0}>
        <h4>Не выбран пользователь</h4>
      </div>
    )}
    return(
      <div className="user-editor col s8" key={"user"+this.props.user.id}>
        <h4>{this.props.user.name.length>0?this.props.user.name:"Новый пользователь"}</h4>
        <form onSubmit={this.handleSubmit}>
          <input defaultValue={this.props.user.id} type="hidden" ref="id-input" name="id"></input>
          <label htmlFor="name">Имя пользователя</label>
          <input defaultValue={this.props.user.name} type="text" ref="name-input" name="name"></input>
          <label htmlFor="email">Email</label>
          <input defaultValue={this.props.user.email} type="text" ref="email-input" name="email"></input>
          <p><label><input defaultChecked={this.props.user.active} type="checkbox" ref="active-input" name="active" /><span>Активен</span></label></p>
          <label htmlFor="password">Пароль</label>
          <div className="password-field"><i className={this.state.hide_password?"fas fa-eye":"fas fa-eye-slash"} onClick={this.hidePassword}></i><input defaultValue={this.props.user.password} type={this.state.hide_password?"password":"text"} ref="password-input" name="password"></input></div>
          <label htmlFor="role">Уровень доступа</label>
          <select defaultValue={this.props.user.role} ref="role-input" name="role">
            <option value="3">Пользователь</option>
            <option value="2">Редактор</option>
            <option value="1">Модератор</option>
          <option value="4">Администратор свалки</option>
          </select>
          <button className="blue-grey waves-effect hoverable waves-light btn" type="submit">{this.state.sent?<i className="fas fa-check"></i>:<i class="fas fa-user-check"></i>}Сохранить</button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = (state,props)=>{
  return({
    user: selectors.user.getCurrentUser(state,props)
  })
}

const mapDispatchToProps = {
  submitUser
}

export default connect(mapStateToProps, mapDispatchToProps)(UserEditor);
