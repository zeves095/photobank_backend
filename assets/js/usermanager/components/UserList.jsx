import React from 'react';
import {connect} from 'react-redux';
import {addUser,chooseUser} from '../actionCreator';
import selectors from '../selectors';

/**
 * Компонент для отображения списка существующих пользователей
 */
export class UserList extends React.Component {

  /**
   * Конструктор компонента
   */
  constructor(props) {
    super(props);
    this.state = {
      "show_inactive": false
    }
  }

  /**
   * Показать/скрыть список неактивных пользователей
   */
  showInactive=()=>{
    this.setState({
      "show_inactive":!this.state.show_inactive
    });
  }

  render() {
    let usersActive = this.props.users_active.map((user)=>{
      return <div key={"userl"+user.id} className={"user-item card-panel white-text "+(this.props.user&&this.props.user.id==user.id?"light-blue accent-3":"blue-grey lighten-2")} onClick={()=>{this.props.chooseUser(user.id)}}><i className="fas fa-user"></i>{user.name}</div>
    });
    let usersInactive = this.props.users_inactive.map((user)=>{
      return <div key={"userl"+user.id} className={"user-item inactive card-panel "+(this.props.user&&this.props.user.id==user.id?"light-blue accent-1":"blue-grey lighten-4")} onClick={()=>{this.props.chooseUser(user.id)}}><i className="fas fa-user-slash"></i>{user.name}</div>
    });
    return(
      <div className="user-list col s4">
        <h4>Пользователи</h4>
        {usersActive}
        <button className="blue-grey waves-effect hoverable waves-light btn" type="button" onClick={this.showInactive}>{this.state.show_inactive?"Скрыть неактивных":"Показать неактивных"}</button>
        <div className={"users-inactive" + (this.state.show_inactive?"":" hidden")}>
          {usersInactive}
        </div>
        <button className="blue-grey waves-effect hoverable waves-light btn" type="button" onClick={()=>{this.props.addUser(this.props.users)}}><i className="fas fa-user-plus"></i>Добавить</button>
      </div>
    );
  }
}

const mapStateToProps = (state,props)=>{
  return({
    user: selectors.user.getCurrentUser(state,props),
    users: selectors.user.getUsers(state,props),
    users_active: selectors.user.getActiveUsers(state,props),
    users_inactive: selectors.user.getInactiveUsers(state,props),
  })
}

const mapDispatchToProps = {
  addUser,
  chooseUser
}

export default connect(mapStateToProps, mapDispatchToProps)(UserList);
