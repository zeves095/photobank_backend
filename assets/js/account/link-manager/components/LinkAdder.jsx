import React, {Component} from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import LinkAddForm from './LinkAddForm';
import LinkResource from './LinkResource';
import ResourceExplorer from './ResourceExplorer';

import { stopEditing } from '../actionCreator'

/**
 * Правые две колонки интерфейса добавления ссылок, включает в себя поиск ресурсов и форму добавления ссылки
 */
export class LinkAdder extends React.Component{

  /**
   * Конструктор компонента
   * @param {Object} props Входные данные из коннекта Redux
   */
  constructor(props){
    super(props);
    this.state={};
  }

  /**
   * Обработчик скрытия этой части интерфейса
   */
  handleCloseAdder = ()=>{
    this.props.stopEditing();
  }

  render(){
    let link_adder = (
      <div className={"link-adder__inner "+(this.props.link_adding?"open":"")}>
        <div className="component-header component-header--subcomponent adder-controls">
          <div className="button-block"><button type="button" name="button" onClick={this.handleCloseAdder}><i className="fas fa-times-circle"></i></button></div>
        </div>
      <div className="component-body__bottom-section resource-section">
        <ResourceExplorer />
      </div>
      <div className="component-body__top-section link-section">
        <h2 className="component-title">
          Добавление ссылки
        </h2>
        <LinkAddForm />
        <LinkResource />
      </div>
    </div>
    )
    return (
      <div className={"link-adder"+(this.props.link_adding?" open":"")}>
        <div className="component-body">
          {this.props.link_adding?link_adder:null}
        </div>
      </div>
    );
  }

}

const mapStateToProps = (state) =>{
  return {
    link_adding: state.link.link_adding,
  }
}

const mapDispatchToProps = {
  stopEditing
}

export default connect(mapStateToProps, mapDispatchToProps)(LinkAdder);
