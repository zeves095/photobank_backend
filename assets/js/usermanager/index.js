import React from 'react';
import ReactDOM from 'react-dom';
import utility from '../photobank/services/UtilityService';

import { UserManagerWrapper } from './components/UserManagerWrapper';

var wrapper = document.getElementById('usermanager-wrapper');
global.config = JSON.parse(wrapper.dataset.config);
wrapper.dataset.config="";

if(typeof UserManagerWrapper != 'undefined') ReactDOM.render(<UserManagerWrapper />, wrapper);
