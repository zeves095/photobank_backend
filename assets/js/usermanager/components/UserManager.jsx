import React from 'react';
import {connect} from 'react-redux';
import UserList from './UserList';
import UserEditor from './UserEditor';
import UserService from '../services/UserService';
import {NotificationService} from '../../services/NotificationService';
import {fetchUsers} from '../actionCreator';
import selectors from '../selectors';

/**
 * Интерфейс работы с поьзователями
 */
export class UserManager extends React.Component {

  /**
   * Конструктор компонента
   * users - Список существующих пользователей
   * current_user - Пользователь, который редактируется в данный момент
   */
  constructor(props) {
    super(props);
  }

  componentWillMount(){
    this.props.fetchUsers();
  }

  render() {
    return(
      <div className="user-manager-main row">
        <div id="notification-overlay">

        </div>
      <nav className="nav-wrapper blue-grey darken-2">
        <div className="brand-logo"><i className="fas fa-camera"></i>PhotoBank - Редактор пользователей</div>
        </nav>
        <UserList />
        <UserEditor />
      </div>
    );
  }
}

const mapStateToProps = (state,props)=>{
  return({

  })
}

const mapDispatchToProps = {
  fetchUsers,
}

export default connect(mapStateToProps, mapDispatchToProps)(UserManager);
