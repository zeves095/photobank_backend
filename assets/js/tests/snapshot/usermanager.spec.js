import React from 'react';
import { UserEditor } from '../../usermanager/components/UserEditor';
import { UserManager } from '../../usermanager/components/UserManager';
import { UserList } from '../../usermanager/components/UserList';
import Enzyme, {render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {store} from '../../usermanager/store/';
import {Provider} from 'react-redux';

Enzyme.configure({adapter: new Adapter()});

describe('snapshot testing', ()=>{
    it('UserEditor рендерится правильно', () => {
      const tree = render(<Provider store={store}><UserEditor /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('UserManager рендерится правильно', () => {
      const tree = render(<Provider store={store}><UserManager fetchUsers={()=>{}} /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
    it('UserList рендерится правильно', () => {
      const tree = render(<Provider store={store}><UserList users_active={[]} users_inactive={[]} /></Provider>).html();
      expect(tree).toMatchSnapshot();
    });
  }
)
