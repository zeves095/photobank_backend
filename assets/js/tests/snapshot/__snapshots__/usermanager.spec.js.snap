// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing UserEditor рендерится правильно 1`] = `"<h4>&#x41D;&#x435; &#x432;&#x44B;&#x431;&#x440;&#x430;&#x43D; &#x43F;&#x43E;&#x43B;&#x44C;&#x437;&#x43E;&#x432;&#x430;&#x442;&#x435;&#x43B;&#x44C;</h4>"`;

exports[`snapshot testing UserList рендерится правильно 1`] = `"<h4>&#x41F;&#x43E;&#x43B;&#x44C;&#x437;&#x43E;&#x432;&#x430;&#x442;&#x435;&#x43B;&#x438;</h4><button class=\\"blue-grey waves-effect hoverable waves-light btn\\" type=\\"button\\">&#x41F;&#x43E;&#x43A;&#x430;&#x437;&#x430;&#x442;&#x44C; &#x43D;&#x435;&#x430;&#x43A;&#x442;&#x438;&#x432;&#x43D;&#x44B;&#x445;</button><div class=\\"users-inactive hidden\\"></div><button class=\\"blue-grey waves-effect hoverable waves-light btn\\" type=\\"button\\"><i class=\\"fas fa-user-plus\\"></i>&#x414;&#x43E;&#x431;&#x430;&#x432;&#x438;&#x442;&#x44C;</button>"`;

exports[`snapshot testing UserManager рендерится правильно 1`] = `"<div id=\\"notification-overlay\\"></div><nav class=\\"nav-wrapper blue-grey darken-2\\"><div class=\\"brand-logo\\"><i class=\\"fas fa-camera\\"></i>PhotoBank - &#x420;&#x435;&#x434;&#x430;&#x43A;&#x442;&#x43E;&#x440; &#x43F;&#x43E;&#x43B;&#x44C;&#x437;&#x43E;&#x432;&#x430;&#x442;&#x435;&#x43B;&#x435;&#x439;</div></nav><div class=\\"user-list col s4\\"><h4>&#x41F;&#x43E;&#x43B;&#x44C;&#x437;&#x43E;&#x432;&#x430;&#x442;&#x435;&#x43B;&#x438;</h4><button class=\\"blue-grey waves-effect hoverable waves-light btn\\" type=\\"button\\">&#x41F;&#x43E;&#x43A;&#x430;&#x437;&#x430;&#x442;&#x44C; &#x43D;&#x435;&#x430;&#x43A;&#x442;&#x438;&#x432;&#x43D;&#x44B;&#x445;</button><div class=\\"users-inactive hidden\\"></div><button class=\\"blue-grey waves-effect hoverable waves-light btn\\" type=\\"button\\"><i class=\\"fas fa-user-plus\\"></i>&#x414;&#x43E;&#x431;&#x430;&#x432;&#x438;&#x442;&#x44C;</button></div><div class=\\"user-editor col s8\\"><h4>&#x41D;&#x435; &#x432;&#x44B;&#x431;&#x440;&#x430;&#x43D; &#x43F;&#x43E;&#x43B;&#x44C;&#x437;&#x43E;&#x432;&#x430;&#x442;&#x435;&#x43B;&#x44C;</h4></div>"`;
